### ISSUE TITLE
IP Address GraphQL Transformer

### ISSUE THREAD

**Issue by hisham** [aws-amplify/amplify-cli#5805]
**Is your feature request related to a problem? Please describe.**
In some cases, it's helpful to log the ip address a mutation came from. 

**Describe the solution you'd like**
Some kind of graphql directive like so: 

`ip: AWSIPAddress @ip_address`

**Describe alternatives you've considered**
Build my own transformer, call a rest api / lambda that does this for now, or just skip this feature for now.

**Comment by akshbhu** [comment]
Hi @hisham 

I am not sure on this but info you can also also get IP address in $context.Identity or $context.request object?
$context object contain sourceIPs as mentioned here : https://docs.aws.amazon.com/appsync/latest/devguide/resolver-context-reference.html

**Comment by hisham** [comment]
@akshbhu yes that's correct. What I'm saying is I want to save ip address as another column in the dynamodb entry. Use case is public API to submit some content to us, would be not a bad idea to store ip address alongside the submitted content in case we need to debug something or identify bots (though you can argue WAF can be used here). This is a nice to have currently. But seems to be a common use case in web apps to store ip address to the database sometimes. 

So the transformer would get the user's ip address from $context and save it to the specified column name.

**Comment by hisham** [comment]
Also once Cloudfront geolocation headers are added to AppSync (https://github.com/awslabs/aws-mobile-appsync-sdk-js/issues/584), might be helpful to have ip geo location address saved to the db as well.

**Comment by mjza** [comment]
same issue here.

**Comment by hisham** [comment]
Another request for this is to ability to save user agent string of the client that did the request.
